lib/segment/src/vector_storage/memmap_vector_storage.rs

```diff
@@ -212,11 +212,12 @@ impl VectorStorage for MemmapVectorStorage {
 
     fn score_points(
         &self,
-        vector: &Vec<VectorElementType>,
+        vector: &[VectorElementType],
         points: &mut dyn Iterator<Item = PointOffsetType>,
         top: usize,
     ) -> Vec<ScoredPointOffset> {
-        let preprocessed_vector = self.metric.preprocess(vector.clone());
```

Comment:
Used to be 1 copy on call and 1 more copy in the case of ``CosineMetric``. Now it is not copying for ``EuclidMetric`` and ``DotProductMetric`` and only 1 copy for ``CosineMetric`` (inside it's ``preprocess``)